[CORE-9493] [CORE-9492] [CORE-9494] dt/dl: Use progress-oriented wait for datalake translation

oleiman opened the pull request:
Some datalake tests fail very occasionally due to transient errors (e.g. uploads failing for a bit). Adding a progress check should make us a bit more resilient.

<!--
See https://github.com/redpanda-data/redpanda/blob/dev/CONTRIBUTING.md#pull-request-body
for more details and examples of what is expected in a PR body.

Content in this top section is REQUIRED. Describe, in plain language, the motivation
behind the change (bug fix, feature, improvement) in this PR and how the included
commits address it.

Add the GitHub keyword `Fixes` to link to bug(s) this PR will fix, e.g.
  Fixes #ISSUE-NUMBER, Fixes #ISSUE-NUMBER, ...

If this PR is a backport, link to the original with `Backport of PR`, e.g.
  Backport of PR #PR-NUMBER
-->

## Backports Required

<!-- Checking at least one of the checkboxes is REQUIRED if this PR is not a backport. -->

- [ ] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [x] v25.3.x
- [ ] v25.2.x
- [ ] v25.1.x
- [ ] v24.3.x

## Release Notes

* none

<!--
If the changes in this PR do not need to be mentioned in the release
notes, then don't add a sub-section and simply list `none`, e.g.

* none

Otherwise, adding a sub-section or `none` is REQUIRED if the PR is not a backport PR.
If this is a backport PR, adding contents to this section will override
the release notes section inherited from the original PR to dev.

Add one or more of the sub-sections with a short description bullet
point of the change, e.g.

### Bug Fixes

* Short description of the bug fix if this is a PR to `dev` branch.

### Features

* Short description of the feature. Explain how to configure.

### Improvements

* Short description of how this PR improves existing behavior.

-->

oleiman committed:
dt/dl: Use progress-oriented wait for datalake translation

Some datalake tests fail very occasionally due to transient errors (e.g.
uploads failing for a bit). Adding a progress check should make us a bit more
resilient.

Signed-off-by: Oren Leiman <oren.leiman@redpanda.com>

oleiman commented:
/ci-repeat 1

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/74957-->
<!--3d0e704d-e5cf-4f93-ac89-ac97760714b6-->
<!--title-->
## Retry command for [Build#74957](https://buildkite.com/redpanda/redpanda/builds/74957)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/tls_metrics_test.py::TLSMetricsTest.test_labels
```
<!--/body-->

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--7c3f741a-ab03-444d-9385-02d4eb160764-->
<!--TEST RESULTS-->
<!--1e37a292-a5fe-49ca-b404-bdb1f2b29547-->
<!--TEST RESULTS-->
<!--035bfcf5-1f61-46a9-8511-4c78339f6d8a-->
<!--TEST RESULTS-->
<!--10e427aa-7820-4504-9e7b-b45059ebbed7-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/74957">build#74957</a></summary>


| test_class                | test_method                                  | test_arguments        | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  | test_history                                                                                                                                                             |
|:--------------------------|:---------------------------------------------|:----------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| EndToEndCloudTopicsTxTest | test_write                                   | null                  | integration | https://buildkite.com/redpanda/redpanda/builds/74957#019a178b-e8a0-4036-9eb9-b6c74972e25e | FLAKY         | 19/21    | upstream reliability is '98.07692307692307'. current run reliability is '90.47619047619048'. drift is 7.60073 and the allowed drift is set to 50. The test should PASS  | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=EndToEndCloudTopicsTxTest&test_method=test_write                       |
| SegmentMsTest             | test_segment_rolling_with_retention_consumer | null                  | integration | https://buildkite.com/redpanda/redpanda/builds/74957#019a178b-e8ac-4726-8d35-2e49efb64ea8 | FLAKY         | 17/21    | upstream reliability is '92.806484295846'. current run reliability is '80.95238095238095'. drift is 11.8541 and the allowed drift is set to 50. The test should PASS    | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=SegmentMsTest&test_method=test_segment_rolling_with_retention_consumer |
| PartitionMovementTest     | test_empty                                   | {"num_to_upgrade": 0} | integration | https://buildkite.com/redpanda/redpanda/builds/74957#019a1783-8c67-4583-8b67-5dd08cd8b920 | FLAKY         | 15/21    | upstream reliability is '93.55322338830585'. current run reliability is '71.42857142857143'. drift is 22.12465 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=PartitionMovementTest&test_method=test_empty                           |
| TLSMetricsTest            | test_labels                                  | null                  | integration | https://buildkite.com/redpanda/redpanda/builds/74957#019a1783-8c59-4dbf-8d50-473225229049 | FLAKY         | 2/21     | upstream reliability is '68.21086261980831'. current run reliability is '9.523809523809524'. drift is 58.68705 and the allowed drift is set to 50. The test should FAIL | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=TLSMetricsTest&test_method=test_labels                                 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/76253">build#76253</a></summary>


| test_class         | test_method                                  | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                 | test_history                                                                                                                                                             |
|:-------------------|:---------------------------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:-----------------------------------------------------------------------------------------------------------------------------------------------------------------------|:-------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| SegmentMsTest      | test_segment_rolling_with_retention_consumer | null             | integration | https://buildkite.com/redpanda/redpanda/builds/76253#019a7f19-d811-43b6-937e-5ef2fcc19d99 | FLAKY         | 19/21    | upstream reliability is '92.66862170087975'. current run reliability is '90.47619047619048'. drift is 2.19243 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=SegmentMsTest&test_method=test_segment_rolling_with_retention_consumer |
| RpkDebugBundleTest | test_debug_bundle                            | null             | integration | https://buildkite.com/redpanda/redpanda/builds/76253#019a7f16-c1f3-4e84-8af3-0264eff46720 | FLAKY         | 20/21    | upstream reliability is '99.75186104218362'. current run reliability is '95.23809523809523'. drift is 4.51377 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=RpkDebugBundleTest&test_method=test_debug_bundle                       |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/76409">build#76409</a></summary>


| test_class                         | test_method                        | test_arguments                                         | test_kind   | job_url                                                                                   | test_status   | passed   | reason                                                                                                                                                                  | test_history                                                                                                                                                                        |
|:-----------------------------------|:-----------------------------------|:-------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| AuditLogTestKafkaApi               | test_no_auth_enabled               | {"audit_transport_mode": "kclient"}                    | integration | https://buildkite.com/redpanda/redpanda/builds/76409#019a8551-04f1-4511-9787-30b3e4e43709 | FLAKY         | 14/21    | upstream reliability is '96.06741573033707'. current run reliability is '66.66666666666666'. drift is 29.40075 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=AuditLogTestKafkaApi&test_method=test_no_auth_enabled                             |
| ClusterLinkingTopicSyncingWithMtls | test_topic_syncing_include_exclude | null                                                   | integration | https://buildkite.com/redpanda/redpanda/builds/76409#019a8556-39fb-4cfd-9720-e2614dfab628 | FLAKY         | 18/21    | upstream reliability is '100.0'. current run reliability is '85.71428571428571'. drift is 14.28571 and the allowed drift is set to 50. The test should PASS             | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ClusterLinkingTopicSyncingWithMtls&test_method=test_topic_syncing_include_exclude |
| DatalakeCustomPartitioningTest     | test_many_partitions               | {"catalog_type": "rest_jdbc", "cloud_storage_type": 1} | integration | https://buildkite.com/redpanda/redpanda/builds/76409#019a8556-39fc-446f-a1ef-486295878f02 | FLAKY         | 2/21     | upstream reliability is '100.0'. current run reliability is '9.523809523809524'. drift is 90.47619 and the allowed drift is set to 50. The test should FAIL             | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=DatalakeCustomPartitioningTest&test_method=test_many_partitions                   |
| WriteCachingFailureInjectionTest   | test_unavoidable_data_loss         | null                                                   | integration | https://buildkite.com/redpanda/redpanda/builds/76409#019a8551-04f4-4e96-ad04-39c764f708e1 | FLAKY         | 20/21    | upstream reliability is '94.83870967741936'. current run reliability is '95.23809523809523'. drift is -0.39939 and the allowed drift is set to 50. The test should PASS | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=WriteCachingFailureInjectionTest&test_method=test_unavoidable_data_loss           |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/76441">build#76441</a></summary>


| test_class                             | test_method                        | test_arguments   | test_kind   | job_url                                                                                   | test_status   | passed   | reason   | test_history                                                                                                                                                                            |
|:---------------------------------------|:-----------------------------------|:-----------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|:---------|:----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
| ClusterLinkingTopicSyncingWithTlsFiles | test_topic_syncing_include_exclude | null             | integration | https://buildkite.com/redpanda/redpanda/builds/76441#019a9049-c0ec-49a0-a226-c946f9073fd4 | FAIL          | 0/1      |          | https://redpanda.metabaseapp.com/dashboard/87-tests?tab=142-dt-individual-test-history&test_class=ClusterLinkingTopicSyncingWithTlsFiles&test_method=test_topic_syncing_include_exclude |
</details>
<!--/body-->

dotnwat commented:
we're doing derivities now?

oleiman commented:
> we're doing derivities now?

lol. iiuc each item is just the latest from each of n live query engines where n is usually 1. so the sum should be "we made progress from the perspective of at least one QE"...unless i misread the python 😅

dotnwat commented:
> > we're doing derivities now?
> 
> lol. iiuc each item is just the latest from each of n live query engines where n is usually 1. so the sum should be "we made progress from the perspective of at least one QE"...unless i misread the python 😅

hah i was just referring to `dt/dl`

oleiman commented:
> hah i was just referring to `dt/dl`

oh hah. d tape over d lake

oleiman commented:
/ci-repeat 3
release
skip-redpanda-build
skip-units
dt-repeat 30
tests/rptest/tests/schema_evolution_test.py::SchemaEvolutionE2ETests.test_old_schema_writer

oleiman commented:
/ci-repeat 3
release
skip-redpanda-build
skip-units
dt-repeat 30
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_old_schema_writer

oleiman commented:
/cdt
dt-repeat=30
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_old_schema_writer

oleiman commented:
/ci-repeat 1

oleiman commented:
/cdt
dt-repeat=3
tests/rptest/tests/datalake/schema_evolution_test.py

oleiman commented:
/ci-repeat 1
release
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_reorder_columns

oleiman commented:
/ci-repeat 1
release
skip-units
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_reorder_columns

oleiman commented:
/cdt
dt-repeat=3
tests/rptest/tests/datalake/schema_evolution_test.py

oleiman commented:
/ci-repeat 1
release
skip-units
tests/rptest/tests/datalake/schema_evolution_test.py::SchemaEvolutionE2ETests.test_reorder_columns

oleiman commented:
/cdt
dt-repeat=3
tests/rptest/tests/datalake/schema_evolution_test.py

oleiman commented:
/ci-repeat 1
release
skip-redpanda-build
skip-units
tests/rptest/tests/datalake/schema_evolution_test.py

andrwng commented on tests/rptest/tests/datalake/datalake_services.py:
```diff
@@ -331,25 +334,35 @@ def wait_for_translation(
+            check=total_count,
+            condition=translation_done,
+            timeout_sec=timeout * 3,
+            progress_sec=timeout,
```
It's probably worth updating the wait_for_translation() args -- it's a bit surprising for the actual timeout to be 3x the input timeout

oleiman replied:
huh good point. sure I had some weird reason for doing that at the time. will update

oleiman replied:
done. updated a few call sites so values make sense.

oleiman replied:
done

oleiman committed:
dt/dl: Update wait_for_translation calls w/ custom timeouts

If some test specifies a non-default timeout for translation:

- extend the timeout
- specify an appropriate progress timeout at the call site

If a test uses defaults, do nothing.

Signed-off-by: Oren Leiman <oren.leiman@redpanda.com>

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/76409-->
<!--fc042c0f-ba7a-4ab9-b5c9-00b18c2b6179-->
<!--title-->
## Retry command for [Build#76409](https://buildkite.com/redpanda/redpanda/builds/76409)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/datalake/custom_partitioning_test.py::DatalakeCustomPartitioningTest.test_many_partitions@{"catalog_type":"rest_jdbc","cloud_storage_type":1}
```
<!--/body-->

oleiman commented:
/ci-repeat 1
debug
dt-repeat=6
tests/rptest/tests/datalake/custom_partitioning_test.py::DatalakeCustomPartitioningTest.test_many_partitions

vbotbuildovich commented:
<!--FAILURES SLASH COMMAND https://buildkite.com/redpanda/redpanda/builds/76441-->
<!--47c89c17-66ff-45fc-9271-5efab65fc9c3-->
<!--title-->
## Retry command for [Build#76441](https://buildkite.com/redpanda/redpanda/builds/76441)
please wait until all jobs are finished before running the slash command
<!--/title-->
<!--body-->
```
/ci-repeat 1
tests/rptest/tests/cluster_linking_topic_syncing_test.py::ClusterLinkingTopicSyncingWithTlsFiles.test_topic_syncing_include_exclude
```
<!--/body-->

oleiman commented:
CI Failures:
- [debug-ducktape](https://buildkite.com/redpanda/redpanda/builds/76441#019a9049-c0ec-49a0-a226-c946f9073fd4) looks like a node crash in a cluster linking test, unrelated to this change

oleiman commented:
/ci-repeat 1
debug
skip-redpanda-build
skip-units
tests/rptest/tests/cluster_linking_topic_syncing_test.py::ClusterLinkingTopicSyncingWithTlsFiles.test_topic_syncing_include_exclude

vbotbuildovich commented:
/backport v25.3.x
